### ISSUE TITLE
.aria and .mid import/export

### ISSUE THREAD

**Issue by ghost** [schismtracker/schismtracker#104]
would be great schismtracker supporting .aria and .mid import/export , even considering information lost in the process - what do you all think about?

**Comment by jangler** [comment]
There actually is some MIDI format code in fmt/mid.c, but I guess it's not used? I would personally love to have MIDI import and export as well, although the significant difference between the MIDI paradigm and the IT paradigm makes import a bit weird, I think…

I don't really know much about .aria files but they sound kind of a niche thing to support compared to MIDI files?

**Comment by sagamusix** [comment]
> There actually is some MIDI format code in fmt/mid.c, but I guess it's not used?

It is, but MID might not be in the default file mask of the module loader (at least here it wasn't). However, it's not configurable by the user, which makes the hopefully intended usecase (importing remix stems or old projects for editing; LISTENING TO MIDI MUSIC IN A TRACKER IS NOT A USECASE ;)) a bit difficult, e.g. you cannot easily change the lines pear beat without recompiling the code.

MIDI export is a bit difficult without duplicating tons of playback code; OpenMPT solves this by temporarily assinging every instrument to an internal "plugin" and then makes use of its already existing plugin architecture to "record" a MIDI file. I guess Schism could do something similar by assinging all instruments to a MIDI channel temporarily and then have a fake MIDI output device which records all those notes.

**Comment by jangler** [comment]
I figured export would only be supported (in a meaningful way) if you already had instruments set up for MIDI; i.e. as you say, have Schism do essentially the same thing it would do for MIDI output but do it to a MIDI file instead.

**Comment by sagamusix** [comment]
Yeah; OpenMPT of course just creates some temporary instruments and deletes them afterwards for its export, since MIDI-based plugins also depend on instruments being present. It's doable, just a lot of work. ;)

**Comment by sagamusix** [comment]
MID files should now show up in the file browser, though this might only affect new installations.
